### PR TITLE
Make launcher library sharable and GAMMARAY_INSTALL_QT_LAYOUT

### PR DESCRIPTION
**Pull request by bog-dan-ro**

### CODE REVIEW

File: common/paths.cpp [hunk 1]
```diff
@@ -51,7 +51,7 @@ void setRootPath(const QString& rootPath)
   Q_ASSERT(QDir(rootPath).exists());
   Q_ASSERT(QDir(rootPath).isAbsolute());
 
-  s_rootPath = rootPath;
+  s_rootPath = QFileInfo(rootPath).absoluteFilePath();
```
Comment (vkrause) [inline common/paths.cpp]: For which case is this needed? The assert in the line above suggests rootPath is already absolute here, no?

File: config-gammaray.h.cmake [hunk 2]
```diff
@@ -3,9 +3,18 @@
 // relative install dirs
 #ifdef Q_OS_ANDROID
 # define GAMMARAY_PLUGIN_INSTALL_DIR "lib"
+#elif defined(GAMMARAY_INSTALL_QT_LAYOUT)
+# define GAMMARAY_PLUGIN_INSTALL_DIR "${PROBE_PLUGIN_INSTALL_DIR}"
```
Comment (vkrause) [inline config-gammaray.h.cmake]: Since this is in the elseif branch of the Android branch above, this will never be hit on Android, is that intended?

Comment (bog-dan-ro) [inline config-gammaray.h.cmake]: Yes.

File: common/paths.cpp (hunk 1 above)
Comment (bog-dan-ro) [inline common/paths.cpp]: I compile it in release mode, I think I found the problem on Android ...

**Comment by vkrause** [comment]
This got integrated by now, in smaller commits.

**Pull request closed without merging**